src/sourmash/minhash.py

```diff
@@ -625,7 +625,7 @@ def set_abundances(self, values, clear=True):
 
             for h, v in values.items():
                 hashes.append(h)
-                abunds.append(v)
+                abunds.append(max(0, v))
```

Comment:
We can't cast negative values (signed int) to unsigned int to be passed to the [rust:add_hash_with_abundance](https://github.com/dib-lab/sourmash/blob/b42a9bedd153374df7851f2658698a4185eebc8a/src/core/src/sketch/minhash.rs#L313). Instead, convert the negative abundance values in Python to zero before passing to rust.